### PR TITLE
#7 Add id reference store

### PR DESCRIPTION
**Pull request by jon-betts**
This is an object for finding, storing and filling out id references.

~The code in here is super rubbish and really needs some love.~ The code got some love and was rebased into oblivion

### CODE REVIEW

**Comment by seanh** [comment]
> The code in here is super rubbish and really needs some love.

It looks totally fine :man_shrugging:

File: h/h_api/bulk_api/id_references.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    created. We can reference this in the group membership with
+    `{"$ref": "my_id"}`.
+
+    It's this classes' job to store and fill out these references.
+    """
+
+    REF_KEY = "$ref"
+
+    def __init__(self):
+        self.id_refs = defaultdict(dict)
```
Comment (seanh) [inline h/h_api/bulk_api/id_references.py]: `self.concrete_ids` might be a better name. They're keyed by id refs, but the point of this dict seems to be to store and retrieve the concrete ids.

Comment (seanh) [reply h/h_api/bulk_api/id_references.py]: Should this have a leading underscore?

Comment (jon-betts) [reply h/h_api/bulk_api/id_references.py]: I've changed this to `_ref_to_concrete`, to let you know what we're mapping from and to.

**Pull request merged**